File: benchmarks/raytracer/raytracer-cpp/src/camera.cpp
#pragma once

#include "vector.cpp"

/*
 * Class holding data about the camera 
 */
class Camera {

	public:
		// position of the camera
		Vector position;
		// coordinates to look at
		Vector lookat;
		// field of view (angle in degrees)
		float fov;

		static Camera create(const Vector *position, const Vector *lookat, const float fov) {
			Camera cam;
			cam.position = *position;
			cam.lookat = *lookat;
			cam.fov = fov;
			return cam;
		}
};

File: benchmarks/raytracer/raytracer-js/src/camera.js

/*
 * Class holding data about the camera 
 */
export class Camera {

	constructor(position, lookat, fov) {
		// position (vector) of the camera
		this.position = position;
		// coordinates (vector) to look at
		this.lookat = lookat;
		// field of view (angle in degrees)
		this.fov = fov;
	}

}